File: NetFramework/C1.Win.Document/VB/SearchText/PasswordForm.vb
Imports System.IO

Public Class PasswordForm

    Public Function EnterPassword(fileName As String) As String
        label2.Text = String.Format("'{0}' is protected. Please enter a Document Open Password.", Path.GetFileName(fileName))
        ActiveControl = tbPassword
        If (ShowDialog() = DialogResult.OK) Then
            Return tbPassword.Text
        End If
        Return Nothing
    End Function

    Public Shared Function DoEnterPassword(fileName As String) As String
        Using f As PasswordForm = New PasswordForm()
            Return f.EnterPassword(fileName)
        End Using
    End Function

End Class

File: Core/FlexReport/CS/ExportFromFlexReportAndPdf/PasswordForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace ExportFromFlexReportAndPdf
{
    public partial class PasswordForm : Form
    {
        public PasswordForm()
        {
            InitializeComponent();
        }

        public string EnterPassword(string fileName)
        {
            label2.Text = string.Format("'{0}' is protected. Please enter a Document Open Password.", Path.GetFileName(fileName));
            ActiveControl = tbPassword;
            if (ShowDialog() == DialogResult.OK)
                return tbPassword.Text;
            return string.Empty;
        }

        public static string DoEnterPassword(string fileName)
        {
            using (PasswordForm f = new PasswordForm())
                return f.EnterPassword(fileName);
        }
    }
}
